src/rmutil/args.h

```diff
@@ -73,6 +77,61 @@ int AC_AdvanceBy(ArgsCursor *ac, size_t by);
  */
 int AC_GetVarArgs(ArgsCursor *ac, ArgsCursor *dest);
 
+typedef enum {
+  AC_ARGTYPE_STRING,
+  AC_ARGTYPE_RSTRING,
+  AC_ARGTYPE_LLONG,
+  AC_ARGTYPE_ULLONG,
+  AC_ARGTYPE_UINT,
+  AC_ARGTYPE_INT,
+  AC_ARGTYPE_DOUBLE,
+  /**
+   * This means the name is a flag and does not accept any additional arguments.
+   * In this case, the target value is assumed to be an int, and is set to
+   * nonzero
+   */
+  AC_ARGTYPE_BOOLFLAG
+} ACArgType;
+
+typedef struct {
+  const char *name;  // Name of the argument
+  void *target;      // [out] Target pointer, e.g. `int*`, `RedisModuleString**`
+  size_t *len;       // [out] Target length pointer. Valid only for strings
+  ACArgType type;    // Type of argument
+  int intflags;      // AC_F_COALESCE, etc.
+} ACArgSpec;
+
+/**
+ * Utilizes the argument cursor to traverse a list of known argument specs. This
+ * function will return:
+ * - AC_OK if the argument parsed successfully
+ * - AC_ERR_ENOENT if an argument not mentioned in `specs` is encountered.
+ * - Any other error is assumed to be a parser error, in which the argument exists
+ *   but did not meet constraints of the type
+ *
+ * Note that ENOENT is not a 'hard' error. It simply means that the argument
+ * was not provided within the list. This may be intentional if, for example,
+ * it requires complex processing.
+ */
+int AC_ParseArgSpec(ArgsCursor *ac, ACArgSpec *specs, ACArgSpec **errSpec);
+
+static inline const char *AC_Strerror(int code) {
+  switch (code) {
+    case AC_OK:
+      return "SUCCESS";
+    case AC_ERR_ELIMIT:
+      return "Value is outside acceptable bounds";
+    case AC_ERR_NOARG:
+      return "Expected an argument, but none provided";
+    case AC_ERR_PARSE:
+      return "Could not convert argument to expected type";
+    case AC_ERR_ENOENT:
+      return "Unknown argument";
+    default:
+      return "(AC: You should not be seeing this message. This is a bug)";
```

Comment:
assert!!

Reply:
Passing a wrong code to a strerror function is not worthy of a program crash.